lib/capybara/queries/text_query.rb

```diff
@@ -42,6 +53,15 @@ def negative_failure_message
       def valid_keys
         COUNT_KEYS + [:wait]
       end
+
+      def visible?
+        @type == :visible or (@type.nil? and Capybara.ignore_hidden_elements)
```

Comment:
I was (and remain) very confused by all the different visibility settings Capy has. When I wrote this line I was looking at the docs for Element#text which only mentions `ignore_hidden_elements` but you're right, `visible_text_only` is looked at too (https://github.com/jnicklas/capybara/blob/master/lib/capybara/node/element.rb#L56).